src/sql/sqlops.proposed.d.ts

```diff
@@ -973,4 +973,31 @@ declare module 'sqlops' {
 		 */
 		export function getProvidersByType<T extends DataProvider>(providerType: DataProviderType): T[];
 	}
+
+	/**
+	 * Context object passed as an argument to command callbacks.
+	 * Defines the key properties required to identify a node in the object
+	 * explorer tree and take action against it.
+	 */
+	export interface ObjectExplorerContext {
```

Comment:
Note: I've opened #1884 to track improving on this design. See the notes there on why this is more cumbersome than we'd like.